topic_popover: Add a 'Default' option and Show 'Unmute' option conditionally.

prakhar1144 opened the pull request:
[CZO Discussion](https://chat.zulip.org/#narrow/stream/101-design/topic/Follow.20Topics.20UI/near/1624043)

The `Unmute` option is available in not-muted streams only when it is the current value.

Add a `Default` option. Available in both muted and not-muted streams. `Default` corresponds to the `INHERIT` visibility policy.

Summary:
* In muted streams:
  * Four options: Mute, Default, Unmute, and Follow.

* In not-muted streams: 
  * Three options: Mute, Default, and Follow.
  * The fourth option, `Unmute` is available only when it is the current value.


<!-- Describe your pull request here.-->

---

<!-- If the PR makes UI changes, always include one or more still screenshots to demonstrate your changes. If it seems helpful, add a screen capture of the new functionality as well.

Tooling tips: https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
-->

**Screenshots and screen captures:**

<details><summary>Popover 1</summary>
<img src="https://github.com/zulip/zulip/assets/56781761/8cdac45a-281b-4ee8-ad2b-82cbf5e0d1ea">
</img>
</details> 

<details><summary>Popover 2</summary>
<img src="https://github.com/zulip/zulip/assets/56781761/2797e584-d6fa-4cf7-b428-5f741a3c8b59">
</img>
</details> 

---

<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [ ] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [x] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [x] End-to-end functionality of buttons, interactions and flows.
- [x] Corner cases, error conditions, and easily imagined bugs.
</details>

prakhar1144 commented on web/templates/topic_sidebar_actions.hbs:
```diff
@@ -11,13 +11,22 @@
             <div class="tabs-container">
-                <div class="tab-option tippy-zulip-tooltip {{#if (eq visibility_policy all_visibility_policies.MUTED)}}selected-tab{{/if}}" data-visibility-policy="{{all_visibility_policies.MUTED}}" data-tippy-content="{{t 'Mute topic' }}" aria-label="{{t 'Mute topic' }}">
+                <div class="tab-option tippy-zulip-tooltip {{#if (eq visibility_policy all_visibility_policies.MUTED)}}selected-tab{{/if}}" data-visibility-policy="{{all_visibility_policies.MUTED}}" data-tippy-content="{{t 'Mute' }}" aria-label="{{t 'Mute' }}">
                     <i class="zulip-icon zulip-icon-mute-new"></i>
```
I have changed the tippy-content from "Mute topic" to "Mute" because "Regular topic" doesn't look good ? (if we plan to keep things consistent).

I think, "Mute", "Unmute", "Follow", "Regular" sounds good and its obvious that we are performing action on the topic itself (as it is a topic popover)

abhijeetbodas2001 replied:
That makes sense to me

abhijeetbodas2001 replied:
Maybe we should change the labels in a separate commit though

prakhar1144 replied:
Done

prakhar1144 commented:
@zulipbot add "mentor review"

abhijeetbodas2001 commented on web/templates/topic_sidebar_actions.hbs:
```diff
@@ -11,13 +11,22 @@
                     <i class="zulip-icon zulip-icon-unmute-new"></i>
                 </div>
-                <div class="tab-option tippy-zulip-tooltip {{#if (eq visibility_policy all_visibility_policies.FOLLOWED)}}selected-tab{{/if}}" data-visibility-policy="{{all_visibility_policies.FOLLOWED}}" data-tippy-content="{{t 'Follow topic' }}" aria-label="{{t 'Follow topic' }}">
+                {{#if stream_muted}}
+                    <div class="tab-option tippy-zulip-tooltip {{#if (eq visibility_policy all_visibility_policies.UNMUTED)}}selected-tab{{/if}}" data-visibility-policy="{{all_visibility_policies.UNMUTED}}" data-tippy-content="{{t 'Unmute' }}" aria-label="{{t 'Unmute' }}">
+                        <i class="zulip-icon zulip-icon-unmute-new"></i>
+                    </div>
+                {{else if (eq visibility_policy all_visibility_policies.UNMUTED)}}
+                    <div class="tab-option tippy-zulip-tooltip selected-tab" data-visibility-policy="{{all_visibility_policies.UNMUTED}}" data-tippy-content="{{t 'Unmute' }}" aria-label="{{t 'Unmute' }}">
```
Maybe we can do this in a single `if` conditional? `if stream_muted or visibility_policy==UNMUTED`.
That would make is very clear in which cases we can possibly show the UNMUTE button (and also deduplicate the inside HTML)

prakhar1144 replied:
Hm, right. Made the changes.

abhijeetbodas2001 commented:
Thanks for the PR! Tested out manually and works great!
Code also looks good to me, apart from the comment above.

prakhar1144 commented:
Updates in the recent push:
* Addressed the review comments.
* Updated the icons as per [CZO discussion.](https://chat.zulip.org/#narrow/stream/101-design/topic/Follow.20Topics.20UI/near/1624597)

abhijeetbodas2001 reviewed (commented):
Thanks for updating! Looks good to me now.
@zulipbot remove "mentor review"
@zulipbot add "integration review"

prakhar1144 commented:
@zulipbot add "integration review"

timabbott commented:
This is great, merged, thanks @prakhar1144 and @abhijeetbodas2001!
